WordPress/Classes/ViewRelated/Post/PostSettings/PostSettingsViewModel.swift

```diff
@@ -154,6 +175,29 @@ final class PostSettingsViewModel: ObservableObject {
         settings.password = selection.password.isEmpty ? nil : selection.password
     }
 
+    // MARK: - Navigation
+
+    func showCategoriesPicker() {
+        let categoriesVC = PostSettingsCategoriesPickerViewController(
+            blog: post.blog,
+            selectedCategoryIDs: settings.categoryIDs
+        ) { [weak self] newSelectedIDs in
+            self?.settings.categoryIDs = newSelectedIDs
+        }
+        viewController?.navigationController?.pushViewController(categoriesVC, animated: true)
```

Comment:
I had to cut some corners and fallback to using UIKit navigation. If I were to wrap `PostSettingsCategoriesPickerViewController` in a SwiftUI view using `UIViewControllerRepresentable`, it wouldn't be able to set the `navigationItems`.